### ISSUE TITLE
[6.4.0] Yarn workspace packages not transpiling

### ISSUE THREAD

**Issue by nandorojo** [martpie/next-transpile-modules#185]
- [x] I HAVE READ [THE FAQ](https://github.com/martpie/next-transpile-modules#faq) AND MY PROBLEM WAS NOT DESCRIBED THERE
- [x] I WILL GIVE 10$ TO CHARITY IF MY SOLUTION WAS ACTUALLY IN THE README/FAQ

lol ^ love that

**Are you trying to transpile a local package or an npm package?**

Local packages. Everything works fine with packages from my `node_modules`. However, other packages in my monorepo don't work.

**Describe the bug**

I upgraded from 4.2.0 (I was using `@module-federation/next-transpile-modules`). My local packages are no longer transpiling. All npm packages transpile just fine.

Once I include local packages, however, I see this:

```sh
next-transpile-modules - an unexpected error happened when trying to resolve "@shared/analytics"
Error: Can't resolve '@shared/analytics' in '/Users/fernandorojo/Developer/madison-green/next-app/applications/speed'
    at getPackageRootDirectory (/Users/fernandorojo/Developer/madison-green/next-app/node_modules/next-transpile-modules/src/next-transpile-modules.js:136:15)
    at Array.map (<anonymous>)
    at withTM (/Users/fernandorojo/Developer/madison-green/next-app/node_modules/next-transpile-modules/src/next-transpile-modules.js:145:34)
    at /Users/fernandorojo/Developer/madison-green/next-app/node_modules/next-compose-plugins/lib/compose.js:100:23
    at Array.forEach (<anonymous>)
    at composePlugins (/Users/fernandorojo/Developer/madison-green/next-app/node_modules/next-compose-plugins/lib/compose.js:77:11)
    at /Users/fernandorojo/Developer/madison-green/next-app/node_modules/next-compose-plugins/lib/index.js:22:38
    at normalizeConfig (/Users/fernandorojo/Developer/madison-green/next-app/node_modules/next/dist/next-server/server/config-shared.js:1:1557)
    at loadConfig (/Users/fernandorojo/Developer/madison-green/next-app/node_modules/next/dist/next-server/server/config.js:8:158)
    at async Span.traceAsyncFn (/Users/fernandorojo/Developer/madison-green/next-app/node_modules/next/dist/telemetry/trace/trace.js:5:584)
    at async /Users/fernandorojo/Developer/madison-green/next-app/node_modules/next/dist/build/index.js:2:127
    at async Span.traceAsyncFn (/Users/fernandorojo/Developer/madison-green/next-app/node_modules/next/dist/telemetry/trace/trace.js:5:584)
error Command failed with exit code 1.
```

**To Reproduce**

I know that an OS issue without a reproduction is really annoying. This will just take a while to reproduce. If my issue isn't clear enough to see a potential fix, please let me know, and I'll get a reproduction made.

**Expected behavior**

I expect my local monorepo folders from inside of `/packages` to resolve properly. 
**Setup**

- Next.js version: `10.1.4-canary.2`
- `next-transpile-modules` version: 
- Node.js version: `v14.15.5`
- `npm`/`yarn` version: `6.4.0`
- Operating System: macOS
- Webpack 4 or 5: I tried with both (using `future.webpack5: true` and with v4). No difference

I'm using yarn workspaces. I have everything setup properly, AFAIK, since it was working well before I upgraded.

Here's some of my code, for reference:

<details>
<summary> <b>next.config.js </b> Click to expand</summary>

```js
const path = require('path')
const fs = require('fs')

const node_modules = path.resolve(__dirname, '../..', 'node_modules')

// these are my local packages, which break
// if I remove this from the scopes array, everything works fine.
const monorepoScope = '@shared'

const scopes = ['@fullcalendar', monorepoScope, '@motify', '@nandorojo']

const scopedPackages = []

fs.readdirSync(node_modules)
  .filter((name) => scopes.includes(name))
  .forEach((scope) => {
    fs.readdirSync(path.resolve(node_modules, scope))
      .filter((name) => !name.startsWith('.'))
      .forEach((folderName) => {
        const { name } = require(path.resolve(
          node_modules,
          scope,
          folderName,
          'package.json'
        ))
        scopedPackages.push(name)
      })
  })

module.exports = require('next-transpile-modules')(
  [
    'dripsy',
    'expo-next-react-navigation',
    ...scopedPackages, 
    'moti',
    'react-doorman',
    'react-native-doorman',
    'sentry-expo',
    '@sentry/node',
    '@sentry/browser',
    '@sentry/integrations',
    'react-native-skeleton-content', 
  ],
  { resolveSymlinks: true } // adding / removing this changes nothing
)
```

</details>

When I `console.log(scopedPackages)`, the names of the local packages are all there:

```js
console.log('[next.config.js] scoped-packages', scopedPackages)

// logs this:
[next.config.js] scoped-packages [
  '@shared/analytics',
  '@shared/api',
  '@shared/api-hooks',
  '@shared/api-services',
  '@shared/auth-components',
  '@shared/auth-hooks', 
  '@shared/calendar-embed',
  '@shared/calendars',
  '@shared/components',
  '@shared/constants',
  '@shared/design',
  '@shared/forms',
  '@shared/helpers',
  '@shared/hooks',
  '@shared/html',
  '@shared/navigation',
  '@shared/next-pages',
  '@shared/payments',
  '@shared/popover',
  '@shared/providers',
  '@shared/queries',
  '@shared/screens',
  '@shared/search',
  '@shared/seo',
  '@shared/theme',
  '@shared/types',
  '@shared/ui',
  '@shared/vendors',
  '@fullcalendar/common',
  '@fullcalendar/core',
  '@fullcalendar/daygrid',
  '@fullcalendar/interaction',
  '@fullcalendar/list',
  '@fullcalendar/luxon',
  '@fullcalendar/moment-timezone',
  '@fullcalendar/react',
  '@fullcalendar/timegrid',
  '@motify/components',
  '@motify/core',
  '@motify/skeleton',
  '@nandorojo/react-native-paper-dates',
  '@nandorojo/react-native-phone-input',
  '@nandorojo/swr-firestore'
]
```

**Additional context**

Thanks for your time here. I maintain a lot of OS projects and I know it's time consuming. This lib has been a big help. Excuse my lack of reproduction, please let me know if this is insufficient info.

**Comment by martpie** [comment]
Have you read the changelog and breaking changes of v5?

**Comment by nandorojo** [comment]
Ah, I checked the FAQ and readme, but forgot the releases page. Looks like some of my "main" fields in my local packages were empty. I added those, and the error is no longer there!

I notice the 5.0.0 release notes say your main path can be an empty file.

I'm doing this now:

```ts
// main: index.ts
export {}
```

If I do this, does that mean the other files inside of this package will still be transpiled? I currently import packages directly from their file, rather than a main entry point, to help with tree shaking.

In other words, will this work?

```js
import Button from '@shared/components/button'
```

Instead of importing from the main entry point of `@shared/components`?

Thanks again for the fast reply @martpie, apologies for missing the release notes.

**Comment by martpie** [comment]
> If I do this, does that mean the other files inside of this package will still be transpiled

Yes, the whole package gets transpiled.

> make sure all your packages have a valid main field.

It's in the `README` as well in the Usage notes ;)

**Comment by mustafa-hanif** [comment]
now will you give $10 to charity?
